peer.go

```diff
@@ -76,9 +76,20 @@ func newPeerList(root *RootPeerList) *PeerList {
 	}
 }
 
-// SetStrategy sets customized peer selection stratedgy.
+// SetStrategy sets customized peer selection strategy.
 func (l *PeerList) SetStrategy(sc ScoreCalculator) {
+	l.Lock()
+	defer l.Unlock()
 	l.scoreCalculator = sc
```

Comment:
Ah that's true, we could have probably just added:

```lang=go
        l.RLock()
        newScore := l.scoreCalculator.GetScore(p)
        l.RUnlock()
```

but... sharing the code will mean that we need to take an outerlock.